dhis2-mobile-program-rules/src/main/java/org/dhis2/mobileProgramRules/RuleEngineExtensions.kt

```diff
@@ -357,6 +358,7 @@ fun ProgramRuleAction.toRuleEngineObject(): RuleAction {
 }
 
 fun ProgramRuleVariable.toRuleVariable(
+    d2: D2,
     attributeRepository: TrackedEntityAttributeCollectionRepository,
     dataElementRepository: DataElementCollectionRepository,
```

Comment:
You are adding D2 to use it in `fetchOptions` but you already have `attributeRepository` and `dataElementRepository` why not add `optionCollectionRepository` instead the full D2 and you already have the three collections?:
Something like:

<img width="741" height="610" alt="Image" src="https://github.com/user-attachments/assets/48edfe28-4982-4635-a723-cca9dcc0ba66" />

What do you think?